airbyte-cdk/python/airbyte_cdk/sources/declarative/async_job/job_orchestrator.py

```diff
@@ -100,9 +101,12 @@ def __init__(
         self._slice_iterator = iter(slices)
         self._running_partitions: List[AsyncPartition] = []
         self._job_tracker = job_tracker
-        self._has_started_a_job = False
         self._exceptions_to_break_on: Tuple[Type[Exception], ...] = tuple(exceptions_to_break_on)
 
+        self._has_started_a_job = False
+        self._has_consumed_every_slice = True
```

Comment:
Ok so there seems to be a bigger issue with the exception logic. The current logic is "If there is an issue during job creation, flag this as `_non_breaking_exceptions`". The logic should be "When we decide we bail out because we have reached the max number of attempts, flag an error (which one?) as `_non_breaking_exceptions`". 

As for why this test does not fail, the tests we had did not fail on the first job by on the second so I'll add one test for that

Reply:
Ok, so I've done a couple things:
* Re-use the same mechanism that retries jobs when they are `FAILED` and `TIMED_OUT` to retry jobs that didn't start too
* Update the test `test_given_exception_when_start_job_and_skip_this_exception` so that the first slice would not have been created and hence the loop would have existed when `self._has_consumed_every_slice = True`.  Not that since we re-use the same mechanism than `FAILED` and `TIMED_OUT` jobs, we don't need `self._has_consumed_every_slice = True` anymore